### DISCUSSION TITLE
Tracing with ozone

### DISCUSSION THREAD

**Discussion by Lucas12138** [apache/ozone#6811] [FAQ] [answered]
Hi community,

Does ozone currently offer any tracing feature? e.g. integration with https://opentelemetry.io/?
It would be really nice if we can have tracing for easier debugging.
Thank you

**Comment by jojochuang** [comment, marked as answer]
Yes we have a basic integration: https://ozone.apache.org/docs/edge/feature/observability.html

i also have a few PRs opened to add more traces.

**Reply by Lucas12138** [reply]
Nice! Thanks @jojochuang 
Just to clarify, for adding more traces, do you mean different kinds of traces or tracing at different places or something else?

**Reply by kerneltime** [reply]
@Lucas12138 extending the current traces to all code paths and capture every latency breakdown at a fine grained level.
